x-pack/plugins/ui_actions_enhanced/public/dynamic_actions/dynamic_action_enhancement.ts

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EnhancementRegistryDefinition } from '../../../../../src/plugins/embeddable/public';
+import { SavedObjectReference } from '../../../../../src/core/types';
+import { SerializableState } from '../../../../../src/plugins/kibana_utils/common';
+import { DynamicActionsState } from '../../../ui_actions_enhanced/public';
+import { UiActionsServiceEnhancements } from '../services';
+
+export const dynamicActionEnhancement = (
+  uiActionsEnhanced: UiActionsServiceEnhancements
+): EnhancementRegistryDefinition => {
+  return {
+    id: 'dynamicActions',
+    telemetry: (state: SerializableState, telemetryData: Record<string, any>) => {
```

Comment:
No strong opinion, just interested - why did you keep the `telemetry` API on the client even if it's not used?

Reply:
so we can keep the same persistable state definition type on both sides